src/providers/krb5/krb5_utils.c

```diff
@@ -570,6 +573,29 @@ parse_krb5_map_user(TALLOC_CTX *mem_ctx, const char *krb5_map_user,
         }
     }
 
+    /* conversion names to fully-qualified names */
+    for (int i = 0; i < size; i++) {
+        fq_name = sss_create_internal_fqname(tmp_ctx,
+                                             name_to_primary[i].id_name,
+                                             dom_name);
+        if (fq_name == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "sss_create_internal_fqname failed\n");
+            goto done;
+        }
+        name_to_primary[i].id_name = talloc_strdup(name_to_primary, fq_name);
```

Comment:
You forgot to check the result of strdup. And why do we need to duplicate the name here after all? Can't we just steal it or assign directly to name_to_primary[i].id_name?